llvm/test/CodeGen/X86/canonicalize-vars.ll

```diff
@@ -0,0 +1,1000 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --default-march x86_64-unknown-linux-gnu --version 5
+; RUN: llc -mattr=sse -mtriple=x86_64 < %s | FileCheck %s -check-prefix=SSE
+; RUN: llc -mattr=sse2 -mtriple=x86_64 < %s | FileCheck %s -check-prefix=SSE2
+; RUN: llc -mattr=+avx -mtriple=x86_64 < %s | FileCheck %s -check-prefix=AVX
+; RUN: llc -mattr=+avx2 -mtriple=x86_64 < %s | FileCheck %s -check-prefix=AVX2
+; RUN: llc -mattr=+avx512f -mtriple=x86_64 < %s | FileCheck %s -check-prefix=AVX512F
+; RUN: llc -mattr=+avx512bw -mtriple=x86_64 < %s | FileCheck %s -check-prefix=AVX512BW
+
+
+define float @canon_fp32_varargsf32(float %a) {
+; SSE-LABEL: canon_fp32_varargsf32:
```

Comment:
Sorry that I am not familiar with X86.
One use case of  canonicalize is to quiet an sNaN.
Will this work for that case?

Reply:
yes that case is handled ``` if (C.isNaN() && C.isSignaling()) {```

Reply:
It seems that it just returns with the value as is.  Will it convert sNaN to qNan?

Reply:
Yes, `mulss 1.0, %a` will convert sNaN to qNaN.

Reply:
Hi @wzssyqa I am not handling that case in this PR, I will raise a subsequent PR to address that.